dashboard/app/controllers/student_snapshots_controller.rb

```diff
@@ -22,4 +22,34 @@ def lessons
 
     render json: {lessons: lessons_data, hasUnnumberedLessons: unit.has_unnumbered_lessons?}
   end
+
+  # GET /student_snapshots/cfu_levels/:lesson_id
+  # Returns all CFU levels from the specified lesson
+  # CFU levels are identified by progression: "Check Your Understanding"
+  def cfu_levels
+    lesson_id = params[:lesson_id]
+    lesson = Lesson.find_by(id: lesson_id)
+    return render json: {error: "Can't find Lesson id=#{lesson_id}"}, status: :bad_request unless lesson
+
+    cfu_levels_data = []
+    lesson.script_levels.each do |script_level|
+      # Check if this script_level has progression: "Check Your Understanding" or "Check For Understanding"
+      if script_level.progression&.match?(/^Check\s+(Your|For)\s+Understanding$/i)
```

Comment:
Is there a way that we can determine whether it's a CFU level based on the level type instead of the progression name?

Level type seems like it might be more consistent than a regex match on the name

Reply:
Yeah, we're sharing the same thaughts on this. It's just that for now it seems to be the only reliable option.
Long term - I think it would be nice to actually set smth like `levelType` or `isCfuLevel` field to the level, as right now the field that we have - don't tied to cfu at all an the fact that levels is a free response, or anything else doesn't guarantee that it's actually a cfu level (which I think is fine, but still worth adding separate field indicating if level is cfu or not)